workspace_host/interface.js

```diff
@@ -250,7 +250,7 @@ async function _getAvailablePort(workspace_id, lowest_usable_port, callback) {
         });
     }
 
-    for (var i = lowest_usable_port; i < 65535; i++) {
+    for (let i = lowest_usable_port; i < 65535; i++) {
```

Comment:
Should we randomly choose ports? Or skip ports we already know are allocated in the DB? While still doing the real check, I mean.

What about locking here? Two `init` requests could conflict and choose the same port. One way to handle this is to have a global variable for available ports, and we take them off before trying them.

Reply:
Working on a better implementation here -> https://github.com/PrairieLearn/PrairieLearn/blob/da422dfe8b853202618931a4d0274544e13a5e37/sprocs/workspace_host_allocate_port.sql

But yes, I think we should consider locking or something to prevent concurrency issues.